humblebundle_downloader/download_library.py

```diff
@@ -310,11 +311,13 @@ def _update_cache_data(self, cache_file_key, file_info):
         # quits it can keep track of the progress
         # Note: Only safe because of single thread,
         # need to change if refactor to multi threading
-        with open(self.cache_file, 'w') as outfile:
+        with open(self.cache_file_temp, 'w') as outfile:
             json.dump(
                 self.cache_data, outfile,
                 sort_keys=True, indent=4,
             )
+            outfile.close() #explicitly close outfile and flush output buffer.
+            os.rename(self.cache_file_temp,self.cache_file) #rename temp file to real file.
```

Comment:
rename should fail if the destination file already exists. I think you need to rename the old file, rename the new file, then delete the old file.